file test failure on freebsd10

```
$ gmake test-file
    JULIA test/file
exception on 1: ERROR: test failed: fname == basename(file)
 in error at error.jl:21
 in default_handler at test.jl:19
 in do_test at test.jl:39
 in test_monitor_wait at file.jl:149
 in runtests at /usr/home/nolta/julia/test/testdefs.jl:5
 in anonymous at multi.jl:660
 in run_work_thunk at multi.jl:621
 in remotecall_fetch at multi.jl:694
 in remotecall_fetch at multi.jl:709
 in anonymous at task.jl:1365
while loading file.jl, in expression starting on line 161
ERROR: test failed: fname == basename(file)
 in error at error.jl:21
 in default_handler at test.jl:19
 in do_test at test.jl:39
 in test_monitor_wait at file.jl:149
 in runtests at /usr/home/nolta/julia/test/testdefs.jl:5
 in anonymous at multi.jl:660
 in run_work_thunk at multi.jl:621
 in remotecall_fetch at multi.jl:694
 in remotecall_fetch at multi.jl:709
 in anonymous at task.jl:1365
while loading file.jl, in expression starting on line 161
while loading /usr/home/nolta/julia/test/runtests.jl, in expression starting on line 35
     * file

gmake[1]: *** [file] Error 1
gmake: *** [test-file] Error 2
```

--- a/test/file.jl
+++ b/test/file.jl
@@ -236,7 +236,11 @@ function test_monitor_wait(tval)
     end
     fname, events = wait(fm)
     close(fm)
-    @test fname == basename(file)
+    if is_linux() || is_windows() || is_apple()
+        @test fname == basename(file)
+    else
+        @test fname == ""  # platforms where F_GETPATH is not available
+    end
     @test events.changed
 end
 

